src/pip/_internal/models/search_scope.py

```diff
@@ -88,20 +88,23 @@ def get_formatted_locations(self) -> str:
                 # exceptions for malformed URLs
                 if not purl.scheme and not purl.netloc:
                     logger.warning(
-                        'The index url "%s" seems invalid, '
-                        'please provide a scheme.', redacted_index_url)
+                        'The index url "%s" seems invalid, ' "please provide a scheme.",
```

Comment:
Should improve this

Reply:
Fixed in latest. :slightly_smiling_face: